AsyncDisplayKit/Layout/ASLayoutSpec.h

```diff
@@ -81,12 +60,9 @@ NS_ASSUME_NONNULL_BEGIN
  */
 @property (nullable, strong, nonatomic) NSArray<id<ASLayoutable>> *children;
 
-/**
- * Returns the child added to this layout spec using the given index.
- *
- * @param index An identifier associated withe the child.
- */
-- (nullable id<ASLayoutable>)childForIndex:(NSUInteger)index;
+@end
+
+@interface ASNullLayoutSpec : ASLayoutSpec
```

Comment:
If we use a singleton, is it possible to eliminate this?  Maybe we could set the singleton to mutable = NO?  It's OK to keep this if having a singleton of a separate class type is valuable (in which case you can add a + method here).

Does it need to be in the public header / what's the main use case for developers?